lib/ClangImporter/ImportDecl.cpp

```diff
@@ -4934,7 +4937,7 @@ namespace {
                                       getOverridableAccessibility(dc),
                                       name, decl->getLoc(), bodyParams,
                                       decl->getLoc(),
-                                      TypeLoc::withoutLoc(elementTy), dc);
+                                      TypeLoc::withoutLoc(elementContextTy),dc);
```

Comment:
This sets the `ElementType`, not the `Type`. The test I added crashes without this fix.